xds/src/main/java/io/grpc/xds/WeightedRoundRobinLoadBalancer.java

```diff
@@ -432,6 +463,83 @@ int pick() {
     }
   }
 
+  // TODO: add javadocs comments
+  @VisibleForTesting
+  static final class StaticStrideScheduler {
+    private Vector<Long> scaledWeights;
+    private int sizeDivisor;
+    private long sequence;
```

Comment:
This should be an AtomicInteger. If you haven't seen https://github.com/grpc/proposal/blob/master/A58-client-side-weighted-round-robin-lb-policy.md#earliest-deadline-first-edf-scheduler , there's a few things there. We don't have to do the _identical_ approach, but we'd avoid doing things differently when unnecessary.

`lambda () -> uint32 next_seq_fn` is "a function you call to get an int32. The equivalent of that in Java is `AtomicInteger.getAndIncrement()`. One important part of that function in the design is that it is shared across scheduler instances. If you re-create the scheduler for the same input data, you get the same results, so the code doesn't have to worry about poor weighting if it is re-created frequently. Since we didn't have that sort of state previously, the old scheduler had to randomize itself.